src/components/parameters/ParameterInfo.svelte

```diff
@@ -0,0 +1,114 @@
+<svelte:options immutable={true} />
+
+<script lang="ts">
+  import InfoIcon from '@nasa-jpl/stellar/icons/info.svg?component';
+  import { createEventDispatcher } from 'svelte';
+  import type { FormParameter, ValueSource } from '../../types/parameter';
+  import ContextMenu from '../context-menu/ContextMenu.svelte';
+  import ValueSourceBadge from './ValueSourceBadge.svelte';
+
+  export let formParameter: FormParameter;
+
+  const dispatch = createEventDispatcher();
+
+  let contextMenu: ContextMenu;
+  let isIconHovered: boolean = false;
+  let isTooltipHovered: boolean = false;
+  let leaveTimeout: NodeJS.Timeout | null = null;
+  let source: ValueSource;
+  let unit: string | undefined = undefined;
+
+  $: if (formParameter) {
+    source = formParameter.valueSource;
+    unit = formParameter.schema?.metadata?.unit?.value;
+  }
+
+  function leaveCallback() {
+    if (leaveTimeout != null) {
+      clearTimeout(leaveTimeout);
+    }
+
+    leaveTimeout = setTimeout(() => {
+      if (isIconHovered || isTooltipHovered) {
+        leaveCallback();
+      } else {
+        contextMenu.hide();
+      }
+    }, 300);
+  }
+
+  function hoverCallback() {
+    if (leaveTimeout != null) {
+      clearTimeout(leaveTimeout);
+    }
+  }
+
+  function onIconOver(e: MouseEvent) {
+    isIconHovered = true;
+    if (!contextMenu.isShown()) {
+      const bounds = (e.target as HTMLElement).getBoundingClientRect();
+      contextMenu.showDirectly(bounds.x + bounds.width, bounds.y, bounds.x);
+    }
+    hoverCallback();
+  }
+
+  function onReset() {
+    dispatch('reset', formParameter);
+  }
+
+  function onTooltipOver() {
+    isTooltipHovered = true;
+    hoverCallback();
+  }
+
+  function onIconOut() {
+    isIconHovered = false;
+    leaveCallback();
+  }
+
+  function onTooltipOut() {
+    isTooltipHovered = false;
+    leaveCallback();
+  }
+</script>
+
+{#if unit || source !== 'none'}
+  <div class="parameter-info-container" role="contentinfo" on:mouseenter={onIconOver} on:mouseleave={onIconOut}>
+    <div><InfoIcon /></div>
```

Comment:
Could we use `color: var(--st-button-icon-color);` for this one? And could this be vertically aligned within the whole parameter row?

Reply:
I can't vertically align this within the parameter row because the parameter could be a struct/array which can expand the height vertically if the row is expanded. This would result in the icon being in a consistent spot relative to each parameter.

Reply:
That being said, I could explicitly bump it up or down a bit if you think it's not centered.

Reply:
Ah right. Yeah maybe just a tiny bit down, looks a bit high.